extension/lib/display.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+(function(){
+var namespace = "lib/display.js";
+if (window[namespace] === true) {
+    return;
+} else {
+    window[namespace] = true;
+}
+
+const avg_read_speed = 200; // WPM
+const avg_letters_per_word = 6; // 4.79; just from Quora, add a space after each word
+
+/*
+Creates a display overlaid on current web page that displays
+the estimated time remaining to read and the current speed of 
+the auto-read mode in WPM. 
+*/
+class Display {
+    /*
+    Initialize a display with total time remaining and initial speed (~200 WPM).
+
+    Parameters:
+    - readableDomIds: string[]. List of dom IDs that contain readable content.
+    - speeed: int. Value proportional to the speed of auto-read mode, before adjustment by sentence length. 
+    */
+    constructor(readableDomIds, speed) {
+        this.readableDomIds = readableDomIds; // Used to calc initial reading time
+        this.html = null;
+        this.time_remaining = this.initTimer(readableDomIds);
+        this.reading_speed = this.initSpeed(speed);
+        this.end = null; 
+
+        this.defineHtml();
+        this.createDisplay(readableDomIds);
+    }
+
+    /*
+    Defines html code for display to be inserted into web page. 
+    */
+    defineHtml() {
+        this.html = `
+            <div id="displayContainer">
+                <div id="timerContainer">
+                    <div id="timerInsideContainer">
+                        <span id="timerNumber">${this.time_remaining}</span> min remaining
+                    </div>
+                </div>
+                <div id="speedContainer">
+                    <div id="speedInsideContainer">
+                        <span id="speedNumber">${this.reading_speed}</span> WPM
+                    </div>
+                </div>
+            </div>
+            `;      
+    }
+
+    /*
+    Inserts display HTML into webpage. 
+    */
+    createDisplay(readableDomIds) {
+        document.getElementById(readableDomIds[0]).innerHTML = this.html;
+        document.getElementById("displayContainer").style.opacity = 1; // For smoother transition
```

Comment:
how does opacity = 1 help w/ smoother transition?

Reply:
Hmm so the idea is that it will come in as opacity=0, then transition to opacity=1 for a smoother effect (see that it starts with opacity=0 in CSS). However, this code appears to not actually do that; apparently 0 and 1 need to be flipped (will update in next pull request)